debezium-connector-mongodb/pom.xml

```diff
@@ -238,13 +191,12 @@
                     <enableAssertions>true</enableAssertions>
                     <systemPropertyVariables>
                         <!-- Make these available to the tests via system properties -->
-                        <connector.mongodb.hosts>${mongo.data.replicaset.name}/${docker.host.address}:27017</connector.mongodb.hosts>
-                        <connector.mongodb.members.auto.discover>false</connector.mongodb.members.auto.discover>
-                        <connector.mongodb.name>mongo1</connector.mongodb.name>
                         <version.mongo.server>${version.mongo.server}</version.mongo.server>
+                        <mongodb.docker.desktop.ports>${mongodb.docker.desktop.ports}</mongodb.docker.desktop.ports>
                         <skipLongRunningTests>${skipLongRunningTests}</skipLongRunningTests>
                     </systemPropertyVariables>
                     <runOrder>${runOrder}</runOrder>
+                    <argLine>--add-opens java.base/java.net=ALL-UNNAMED</argLine>
```

Comment:
@jpechane this is required for `FakeDns` to work reliably on all JDK 11+ (actually 9+ but that doesn't concern us).  Alternatively `FakeDns` can be disabled using `docker.desktop.disable.fake.dns`property, however then users with Docker Desktop will have to modify `/etc/hosts` in such case our test containers will log a warning with instructions  (to the best extend).

Reply:
Unfortunately JDK9 to JDK17 don't provide a way to provide custom implementation of `NameService` without reflective access to private members. A proper SPI to do this was added in JDK18